controller/resources/deployment.yaml

```diff
@@ -2,7 +2,7 @@
 kind: Deployment
 apiVersion: apps/v1
 metadata:
-  name: objectstorage-controller
+  name: controller
```

Comment:
I am hesitant to change the name of deployments to something nondescript while it is being installed into the `default` namespace.

Reply:
Using the following:

https://github.com/kubernetes-sigs/container-object-storage-interface-api/blob/cb5cd45613c87f5fcc6049eba98728e51ac564ca/controller/kustomization.yaml#L4-L5

Makes the rendered manifests named according to the Kustomization:

```yaml
# ...
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: container-object-storage-interface-controller
    app.kubernetes.io/component: controller
    app.kubernetes.io/managed-by: kustomize
    app.kubernetes.io/name: container-object-storage-interface-controller
    app.kubernetes.io/part-of: container-object-storage-interface
  name: container-object-storage-controller
  namespace: container-object-storage-system
# ...
```

I also replaced the namespace, we should not deploy to default namespace, and the controller will live in the new `container-object-storage-system` namespace.